cpp/config/Make.xcframework.rules

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) ZeroC, Inc. All rights reserved.
+
+# Creates a xcframework from a static component and its headers
+# $1: The name of the framework (Ice, IceDiscovery, etc.)
+# $2: The platform
+# $3: Output path
+define create-xcframework-from-library
+
+lib/$1.xcframework: lib/lib$1.a
+	$(E) Creating lib/$1.xcframework
+	$(Q)$(RM) -r lib/$1.xcframework
+	$(Q)if [ -d include/$1 ]; then \
+		xcodebuild -create-xcframework -library lib/lib$1.a -headers include/$1 -output lib/$1.xcframework ; \
+		cp include/generated/$1/*.h lib/$1.xcframework/macos-arm64/Headers ; \
+		mkdir -p lib/$1.xcframework/macos-arm64/Headers/$1 ; \
+		find lib/$1.xcframework/macos-arm64/Headers/* -prune ! -name $1 -exec mv {} lib/$1.xcframework/macos-arm64/Headers/$1 \; ; \
```

Comment:
why do we need this find/mv command?

Reply:
The headers are copied directly into the `Headers` folder. We need them to be in eg. `Headers/Ice/`, so we move them. I tried using module maps instead but was unable to get it to work.